pkg/middleware/auth.go

```diff
@@ -0,0 +1,66 @@
+package middleware
+
+import (
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/rs/zerolog/hlog"
+	"github.com/warrant-dev/warrant/pkg/service"
+)
+
+var warrantApiKey []byte
+
+func init() {
+	warrantApiKey = []byte(os.Getenv("WARRANT_API_KEY"))
```

Comment:
```suggestion
	apiKey = []byte(os.Getenv("API_KEY"))
```

Can we also log a warning here if the key is empty?
```go
log.Info.Msg("Running without an API key. We recommend providing an API key when running in production.")
```